sql/src/main/java/io/crate/expression/scalar/timestamp/TimezoneFunction.java

```diff
@@ -0,0 +1,131 @@
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Locale;
+
+public class TimezoneFunction extends Scalar<Long, Object> {
+
+    public static final String NAME = "timezone";
+    private static final ZoneId UTC = ZoneId.of("UTC");
+    private static final FuncParams FUNCTION_PARAMS = FuncParams.builder(Param.STRING, Param.ANY).build();
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(NAME, new BaseFunctionResolver(FUNCTION_PARAMS) {
+            @Override
+            public FunctionImplementation getForTypes(List<DataType> signatureTypes) throws IllegalArgumentException {
+                DataType returnType = null;
+                boolean paramHadTimezone = false;
+                int timestampParamTypeId = signatureTypes.get(1).id();
+                switch (timestampParamTypeId) {
+                    case LongType.ID:
+                    case TimestampType.ID_WITHOUT_TZ:
+                        returnType = DataTypes.TIMESTAMPZ;
+                        break;
+                    case TimestampType.ID_WITH_TZ:
+                        returnType = DataTypes.TIMESTAMP;
+                        paramHadTimezone = true;
+                        break;
+                    default:
+                        throw new IllegalArgumentException(String.format(
+                            Locale.ENGLISH,
+                            "invalid argument to %s: expected second parameter" +
+                            " to be of either type [%s, %s, %s] but found %s",
+                            NAME,
+                            LongType.INSTANCE.getName(),
+                            TimestampType.INSTANCE_WITH_TZ.getName(),
+                            TimestampType.INSTANCE_WITHOUT_TZ.getName(),
+                            signatureTypes.get(1)));
+                }
+                return new TimezoneFunction(new FunctionInfo(new FunctionIdent(NAME, signatureTypes), returnType),
+                                            paramHadTimezone);
+            }
+        });
+    }
+
+    private final FunctionInfo info;
+    private final boolean paramHadTimezone;
+
+    private TimezoneFunction(FunctionInfo info, boolean paramHadTimezone) {
+        this.info = info;
+        this.paramHadTimezone = paramHadTimezone;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Long evaluate(TransactionContext txnCtx, Input<Object>... args) {
+        assert args.length == 2 : String.format(Locale.ENGLISH,
+                                                "number of arguments must be 2, got %d instead",
+                                                args.length);
+        String zoneStr = (String) args[0].value();
+        if (zoneStr == null || zoneStr.isBlank()) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
```

Comment:
In PostgreSQL a zone with `null` as value results in `null` as result

Reply:
not sure what it returns, it is rendered as blanc, however the return type is a timestamp with/without time zone (depending on the type of the second parameter). I will return null, and NULL will be rendered.

Reply:
You can do something like `\pset null '💩'` in psql to see `null` values in a different format. (Can also be added to `.psqlrc` to have it permanently set.

`zoneStr == null`  should result in `null`

Reply:
alles klar, same for the comment on the timestamp param being null, thank you